cartography/intel/aws/cloudtrail_management_events.py

```diff
@@ -0,0 +1,553 @@
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+@aws_handle_regions
+def get_cloudtrail_events(
+    boto3_session: boto3.Session, region: str, lookback_hours: int = 24
+) -> List[Dict[str, Any]]:
+    """
+    Fetch CloudTrail management events from the specified time period.
+
+    :type boto3_session: boto3.Session
+    :param boto3_session: The boto3 session to use for API calls
+    :type region: str
+    :param region: The AWS region to fetch events from
+    :type lookback_hours: int
+    :param lookback_hours: Number of hours back to retrieve events from
+    :rtype: List[Dict[str, Any]]
+    :return: List of CloudTrail events
+    """
+    client = boto3_session.client(
+        "cloudtrail", region_name=region, config=get_botocore_config()
+    )
+
+    # Calculate time range
+    end_time = datetime.utcnow()
+    start_time = end_time - timedelta(hours=lookback_hours)
+
+    logger.info(
+        f"Fetching CloudTrail management events for region '{region}' "
+        f"from {start_time} to {end_time} ({lookback_hours} hours)"
+    )
+
+    events = []
+
+    try:
+        # Focus on STS events for role assumptions
+        paginator = client.get_paginator("lookup_events")
+
+        page_iterator = paginator.paginate(
+            LookupAttributes=[
+                {"AttributeKey": "EventSource", "AttributeValue": "sts.amazonaws.com"}
+            ],
+            StartTime=start_time,
+            EndTime=end_time,
+            PaginationConfig={
+                "MaxItems": 10000,  # Reasonable limit to prevent excessive API calls
+                "PageSize": 50,  # CloudTrail API limit per page
+            },
+        )
+
+        for page in page_iterator:
+            events.extend(page.get("Events", []))
+
+        logger.info(
+            f"Retrieved {len(events)} CloudTrail management events from region '{region}'"
+        )
+
+    except Exception as e:
+        logger.warning(
+            f"Failed to retrieve CloudTrail management events for region '{region}': {str(e)}"
+        )
+
+    return events
+
+
+@timeit
+def transform_cloudtrail_events_to_role_assumptions(
+    events: List[Dict[str, Any]],
+    region: str,
+    current_aws_account_id: str,
+) -> List[Dict[str, Any]]:
+    """
+    Transform raw CloudTrail events into role assumption relationships.
+
+    :type events: List[Dict[str, Any]]
+    :param events: List of raw CloudTrail events from lookup_events API
+    :type region: str
+    :param region: The AWS region where events were retrieved from
+    :type current_aws_account_id: str
+    :param current_aws_account_id: The AWS account ID being synced
+    :rtype: List[Dict[str, Any]]
+    :return: List of role assumption records in standardized format
+    """
+    role_assumptions = []
+
+    logger.info(
+        f"Transforming {len(events)} CloudTrail events to role assumptions for region '{region}'"
+    )
+
+    for event in events:
+        try:
```

Comment:
Have you run the code and ran into this exception? If not, remove the try+except. Pre-emptive exceptions are not needed for this scenario. Code for the happy path and add exceptions as needed.

Right now, even if this exception does happen, we're writing a log message and swallowing it, and this makes the code harder to trace if that issue comes up. Let the exception fail loudly so that it's easier to trace. Then we can add in the handling if appropriate in quick follow-up PRs.

Reply:
[block] can we remove the try + excepts here?